data/abi/abi_encode.go

```diff
@@ -495,16 +503,31 @@ func ParseArgJSONtoByteSlice(argTypes []string, jsonArgs []string, applicationAr
 		return fmt.Errorf("input argument number %d != method argument number %d", len(jsonArgs), len(abiTypes))
 	}
 
-	// change the input args to be 1 - 14 + 15 (compacting everything together)
-	if len(jsonArgs) > 14 {
-		compactedType, err := MakeTupleType(abiTypes[14:])
+	// Up to 16 app arguments can be passed to app call. First is reserved for method selector,
+	// and the rest are for method call arguments. But if more than 15 method call arguments
+	// are present, then the 14th+ are placed in a tuple in the last app argument slot
```

Comment:
```suggestion
	// are present, then the method arguments after the 14th are placed in a tuple in the last app argument slot
```

Reply:
I don't feel strong here, given that @jannotti previously suggested that we might want to bump the number of max-argument-num from 16 to 32, then all the comments should also be updated later...